### PR TITLE
Added logging with validation extraction and some unit tests

### PR DESCRIPTION
**Pull request by gregito**

### CODE REVIEW

File: lib/services/authenticate.dart [hunk 1]
```diff
@@ -11,21 +12,27 @@ class FireStoreUtils {
...
+    log.logger.d("Getting current user by its id: $uid");
     DocumentSnapshot<Map<String, dynamic>> userDocument =
         await firestore.collection(usersCollection).doc(uid).get();
     if (userDocument.data() != null && userDocument.exists) {
-      return User.fromJson(userDocument.data()!);
+      User user = User.fromJson(userDocument.data()!);
+      log.logger.i("User exists: $user");
+      return user;
     } else {
+      log.logger.i("User does not exists with id: $uid");
```
Comment (TerjekiG) [inline lib/services/authenticate.dart:23]: Is this only an "info" event?

File: lib/services/authenticate.dart [hunk 2]
```diff
@@ -93,20 +110,28 @@ class FireStoreUtils {
...
+          isStartDateSet:
+              const bool.fromEnvironment("REGISTER_WITH_START_DATE_SET"));
+      log.logger
+          .i("About to request user creation using the following data: $user");
       String? errorMessage = await createNewUser(user);
       if (errorMessage == null) {
+        log.logger.i("User (with email: $emailAddress) created!");
         return user;
       } else {
+        log.logger.w("User cannot be created due to: $errorMessage");
```
Comment (TerjekiG) [inline lib/services/authenticate.dart:129]: Is this only "warning" and not "error" because you can try again?

File: lib/ui/auth/login/login_screen.dart
Comment (TerjekiG) [inline lib/ui/auth/login/login_screen.dart:1]: I thought constants get allcaps, turns out, they don't in dart: https://dart.dev/tools/linter-rules/constant_identifier_names

File: test/services/validator_test.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import 'package:kisgeri24/services/validator.dart' as under_test;
+import 'package:test/test.dart';
+
+void main() {
+  group("Testing phone number validation against invalid inputs.", () {
+    Map<String?, String> inputs = {
+      null: "Mobile phone number is required",
+      "": "Mobile phone number is required",
+      "abc1234567": "Mobile phone number must contain only digits",
+      "12": "Mobile phone number must contain only digits",
```
Comment (TerjekiG) [inline test/services/validator_test.dart:10]: Technically, this one only contains digits.

File: lib/services/authenticate.dart (hunk 1 above)
Comment (gregito) [reply lib/services/authenticate.dart:23]: yeah, I usually consider application errors (like exceptions) as a warning-like event. this is just an info that the user does not exists

File: lib/services/validator.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  if ((value?.length ?? 0) < 6) {
+    return 'Password must be more than 5 characters';
+  } else {
+    return null;
+  }
+}
+
+String? validateEmail(String? value) {
+  String pattern =
+      r'^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$';
```
Comment (OnionKiller) [inline lib/services/validator.dart:44]: It may worth to consider to use a package instead of this RegEx. It would probably only make difference in very extreme edge cases. This package seems to be the flutter standard for email validation: [email_validator](https://pub.dev/packages/email_validator).
(Edge cases for email, if someone is interested: [https://stackoverflow.com/a/201378/8659684](https://stackoverflow.com/a/201378/8659684))

File: test/services/validator_test.dart [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+      "0612345678",
+      "06(1)234-5678",
+      "06-1-234-5678",
+      "061234567",
+      "06(12)345-678",
+      "06-12-345-678",
+    };
+    for (var input in inputs) {
+      test(
+          "The given input ($input) is not acceptable as a valid phone number!",
```
Comment (OnionKiller) [inline test/services/validator_test.dart:56]:
```suggestion
          "The given input ($input) is not accepted as a valid phone number!",
```

Comment (OnionKiller) [reply test/services/validator_test.dart:56]: I'm confused, isn't this description parameter be a description?
Later it seems to be used as a description.

File: lib/services/validator.dart (hunk 4 above)
Comment (gregito) [reply lib/services/validator.dart:44]: yeahh, looks neat, gonna replace it with this package. I'm also not a huge fan of robust regexes 😅

File: test/services/validator_test.dart (hunk 5 above)
Comment (gregito) [reply test/services/validator_test.dart:56]: I'd stick to the `acceptable`.  And yeah, this string will be shown if the assertion would fail

Comment (OnionKiller) [reply test/services/validator_test.dart:56]: Ok

File: lib/services/validator.dart (hunk 4 above)
Comment (gregito) [reply lib/services/validator.dart]: updated as promised

**Review by OnionKiller — APPROVED** [review]
lgtm
